### PR TITLE
[DT-934] Create a banner similar to the Terra Banner on DUOS to inform users that chrome is our preferred browser

### PR DESCRIPTION
**Pull request by cinyecai**
### Addresses
https://broadworkbench.atlassian.net/browse/DT-934
### Summary
Displays a warning alert for users opening DUOS on a browser that's not Chrome or Firefox, user cannot click the Sign In button until Alert is closed

https://github.com/user-attachments/assets/08a8e1b3-d60a-40a4-ba87-d2b0057a7aed

**Note:** Updated the appearance of the alert to be less extreme - see comment below

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/outdated-browser-message.js
Comment (pshapiro4broad) [inline src/outdated-browser-message.js]: Does `window.chrome` generate a warning when compiled? It was flagged in the editor. If so, I wonder if there's a good / easy way to fix the warning.

**Review by pshapiro4broad — APPROVED** [review]
Looks good

File: src/outdated-browser-message.js
Comment (cinyecai) [reply src/outdated-browser-message.js]: I don't see any warnings about this particular file when compiled

File: package.json
Comment (fboulnois) [inline package.json]: This package was last published 4 years ago and depends on another package that had multiple major revisions (and potentially security fixes).

I think we should evaluate whether we can do this without depending on an external package, and use our existing notification machinery to warn users.

**Review by fboulnois — DISMISSED** [review]
See below:

File: package.json
Comment (rushtong) [reply package.json]: Agreed on this. Additionally, we generally pin versions explicitly due to historic problems with npm point updates.

File: src/outdated-browser-message.js
Comment (rushtong) [inline src/outdated-browser-message.js]: This style, while it does look like Terra, is extreme. I'd like to ensure we get PM feedback/approval before merging.

File: public/index.html
Comment (rushtong) [inline public/index.html]: Note that `cypress/component-index.html` looks a lot like this file so that component tests can have the same basic index to work with. I would love to see a test for this if possible.

File: package.json
Comment (cinyecai) [reply package.json]: I could probably do it using the Alert already in Duos, figuring out how to detect the browser being used will be the harder part - I'll look into it

Comment (cinyecai) [reply package.json]: @fboulnois @rushtong After some searching, I found an alternative approach could be to detect the browser using the user agent but this seems generally to be recommended **against** for being unreliable. I'm not sure if this would be better or worse than using this package, and I'm struggling to find other straightforward ways to do browser detection. Let me know your thoughts!

Comment (rushtong) [reply package.json]: This looks promising as an alternative: https://github.com/pure-js/browser-detection

**Comment by cinyecai** [comment]
Updated visual:


https://github.com/user-attachments/assets/0e5ffa5f-e506-4200-9ea5-99cc35191ce2

File: package.json
Comment (fboulnois) [inline package.json:17]:
```suggestion
    "browser-version-detection": "2.1.8",
```

We should lock this version down so it doesn't automatically change.

This version hasn't seen [any updates for 7 years](https://www.npmjs.com/package/browser-version-detection), and the implementation is [straightforward](https://github.com/pure-js/browser-detection/blob/main/src/detect-browser-version.ts), so I would continue to advocate not depending on a package to do this and instead using a simple regex for browser detection (which is ultimately what this library and the previous one do).

In general, [browser detection](https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent) is not recommended, although I know we also do this on Terra UI. It might be better to identify the specific features that Safari doesn't support, and long term work to either stop depending on these features or use polyfills for those features that we really need. This is a separate issue from showing the alert, so in the meantime we might still want to show the alert regardless of whether we are doing browser or feature detection.

**Review by fboulnois — COMMENTED** [review]
See below:

File: package.json
Comment (rushtong) [reply package.json:17]: I'm a little confused here ... the latest release version I see is [3.1.0](https://github.com/pure-js/browser-detection/releases) that is from last year. Is there a reason we're looking at 2.1.8 (which I don't even see in the list of releases)? I do see that NPM sees [2.1.8](https://www.npmjs.com/package/browser-version-detection), which is extra confusing.

Comment (rushtong) [reply package.json:17]: Aside from the library version question, I do agree in theory with Florian's feedback. To date, the only issue we've seen with Safari is variations in CSS rendering. I don't think we're using any browser-specific features, so to really tackle that question, we'd need a different kind of investigation effort.

**Pull request closed without merging**
